src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/UseEventsWhereAppropriateTests.cs

```diff
@@ -9,12 +12,525 @@ public class UseEventsWhereAppropriateTests : DiagnosticAnalyzerTestBase
+        [Fact]
+        public void NoDiagnostic_NamingCases()
+        {
+            VerifyCSharp(@"
+using System;
+
+public class EventsClass1
+{
+    public event EventHandler RaiseFileEvent;
+
+    public string RaiseAnotherProperty => null;
+
+    public void SomeMethodThatDoesNotStartWithRaise()
+    {
+    }
+
+    public void RaisedRoutine()
+    {
+    }
+
+    public void AddOneAssembly()
+    {
+    }
+
+    public void Remover()
+    {
+    }
+}
+");
+
+            VerifyBasic(@"
+Imports System
+
+Public Class EventsClass1
+    Public Event RaiseFileEvent As EventHandler
+
+    Public ReadOnly Property RaiseAnotherProperty() As String
+        Get
+            Return Nothing
+        End Get
+    End Property
+
+    Public Sub SomeMethodThatDoesNotStartWithRaise()
+    End Sub
+
+    Public Sub RaisedRoutine()
+    End Sub
+
+    Public Sub AddOneAssembly()
+    End Sub
+
+    Public Sub Remover()
+    End Sub
+End Class
+");
+        }
+
+        [WorkItem(380, "https://github.com/dotnet/roslyn-analyzers/issues/380")]
+        [Fact]
+        public void NoDiagnostic_InterfaceMemberImplementation()
+        {
+            VerifyCSharp(@"
+using System;
+
+public class InterfaceImplementation : I
+{
+    // Explicit interface implementation - Rule does not fire.
+    void I.FireOnSomething_InterfaceMethod1()
+    {
+        throw new NotImplementedException();
+    }
+
+    // Implicit interface implementation - Rule does not fire.
+    public void FireOnSomething_InterfaceMethod2()
+    {
+        throw new NotImplementedException();
+    }
+}
+
+#pragma warning disable CA1030 // We are only testing no violations in InterfaceImplementation in this test, so suppress issues reported in the interface.
+public interface I
+{
+    // Interface methods - Rule fires.
+    void FireOnSomething_InterfaceMethod1();
+    void FireOnSomething_InterfaceMethod2();
+}
+#pragma warning restore CA1030
+");
+
+            VerifyBasic(@"
+Imports System
+
+Public Class InterfaceImplementation
+	Inherits I
+	' Explicit interface implementation - Rule does not fire.
+	Private Sub I_FireOnSomething_InterfaceMethod1() Implements I.FireOnSomething_InterfaceMethod1
+		Throw New NotImplementedException()
+	End Sub
+
+	' Implicit interface implementation - Rule does not fire.
+	Public Sub FireOnSomething_InterfaceMethod2() Implements I.FireOnSomething_InterfaceMethod2
+		Throw New NotImplementedException()
+	End Sub
+End Class
+
+#Disable Warning CA1030 ' We are only testing no violations in InterfaceImplementation in this test, so suppress issues reported in the interface.
+Public Interface I
+	' Interface methods - Rule fires.
+	Sub FireOnSomething_InterfaceMethod1() Implements I.FireOnSomething_InterfaceMethod2
+	Sub FireOnSomething_InterfaceMethod2()
+End Interface
+#Enable Warning CA1030
+");
+        }
+
+        [WorkItem(380, "https://github.com/dotnet/roslyn-analyzers/issues/380")]
```

Comment:
For Roslyn, we did both - tests for new features are generally tagged with the work item tracking the feature work and then if a test gets broken with bug(s), we replace the work item with the bug number.